test/model.jl

```diff
@@ -492,6 +515,85 @@ function test_copy_direct_mode()
     @test_throws ErrorException JuMP.copy(model)
 end
 
+function test_copy_filter()
+    copy_model_style_mode(true, MOIU.AUTOMATIC, true)
+end
+    
+function test_copy_filter_array()
+    model = Model()
+    model.optimize_hook = dummy_optimizer_hook
```

Comment:
The optimize hook and extension data aren't really relevant for these tests. The same functionality is already tested in copy_model_style_mode. Try to keep the tests focused on one thing, if possible (not all JuMP tests do this).